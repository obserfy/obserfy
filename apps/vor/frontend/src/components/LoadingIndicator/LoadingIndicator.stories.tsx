import { FC } from "react"
import LoadingIndicator from "./LoadingIndicator"

export default {
  title: "Core/LoadingIndicator",
  component: LoadingIndicator,
  parameters: {
    componentSubtitle: "Just a simple LoadingIndicator",
  },
}

export const Basic: FC = () => <LoadingIndicator size={24} />
