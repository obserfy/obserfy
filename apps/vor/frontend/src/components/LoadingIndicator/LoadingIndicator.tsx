// @ts-nocheck
import { keyframes } from "@emotion/react"
import { FC } from "react"
import { Box, BoxProps } from "theme-ui"

const ellipsis1 = keyframes(`
 0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
`)

const ellipsis2 = (gapSize: number): string =>
  keyframes(`
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(${gapSize * 3}px, 0);
  }
`)

const ellipsis3 = keyframes(`
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
`)

interface Props extends BoxProps {
  size?: number
}
export const LoadingIndicator: FC<Props> = ({
  color = "text",
  size = 18,
  ...props
}) => {
  const ballSize = Math.round(size / 6)
  const gapSize = Math.round(size / 10)

  return (
    <Box
      mx={1}
      sx={{
        display: "inline-block",
        position: "relative",
        width: `${size}px`,
        height: `${size}px`,
        "& div": {
          position: "absolute",
          top: `${Math.round((size - ballSize) * 0.5)}px`,
          width: `${ballSize}px`,
          height: `${ballSize}px`,
          borderRadius: "50%",
          backgroundColor: color,
          animationTimingFunction: "cubic-bezier(0, 1, 1, 0)",
          "&:nth-of-type(1)": {
            left: `${gapSize}px`,
            animation: `${ellipsis1} 0.6s infinite`,
          },
          "&:nth-of-type(2)": {
            left: `${gapSize}px`,
            animation: `${ellipsis2(gapSize)} 0.6s infinite`,
          },
          "&:nth-of-type(3)": {
            left: `${gapSize * 4}px`,
            animation: `${ellipsis2(gapSize)} 0.6s infinite`,
          },
          "&:nth-of-type(4)": {
            left: `${gapSize * 7}px`,
            animation: `${ellipsis3} 0.6s infinite`,
          },
        },
      }}
      {...props}
    >
      <div />
      <div />
      <div />
      <div />
    </Box>
  )
}

export default LoadingIndicator
