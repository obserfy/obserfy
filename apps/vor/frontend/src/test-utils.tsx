/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
