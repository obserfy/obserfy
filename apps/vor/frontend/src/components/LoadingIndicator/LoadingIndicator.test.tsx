import { render } from "../../test-utils"
import { Basic } from "./LoadingIndicator.stories"

describe("LoadingIndicator", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
