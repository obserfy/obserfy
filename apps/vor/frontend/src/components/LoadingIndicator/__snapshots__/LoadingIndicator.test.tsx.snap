// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator should render correctly 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: scale(0);
    -moz-transform: scale(0);
    -ms-transform: scale(0);
    transform: scale(0);
  }

  100% {
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: translate(0, 0);
    -moz-transform: translate(0, 0);
    -ms-transform: translate(0, 0);
    transform: translate(0, 0);
  }

  100% {
    -webkit-transform: translate(6px, 0);
    -moz-transform: translate(6px, 0);
    -ms-transform: translate(6px, 0);
    transform: translate(6px, 0);
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: translate(0, 0);
    -moz-transform: translate(0, 0);
    -ms-transform: translate(0, 0);
    transform: translate(0, 0);
  }

  100% {
    -webkit-transform: translate(6px, 0);
    -moz-transform: translate(6px, 0);
    -ms-transform: translate(6px, 0);
    transform: translate(6px, 0);
  }
}

@keyframes animation-2 {
  0% {
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  100% {
    -webkit-transform: scale(0);
    -moz-transform: scale(0);
    -ms-transform: scale(0);
    transform: scale(0);
  }
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: 4px;
  margin-right: 4px;
  display: inline-block;
  position: relative;
  width: 24px;
  height: 24px;
}

.emotion-1 div {
  position: absolute;
  top: 10px;
  width: 4px;
  height: 4px;
  border-radius: 50%;
  background-color: var(--theme-ui-colors-text);
  -webkit-animation-timing-function: cubic-bezier(0, 1, 1, 0);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}

.emotion-1 div:nth-of-type(1) {
  left: 2px;
  -webkit-animation: animation-0 0.6s infinite;
  animation: animation-0 0.6s infinite;
}

.emotion-1 div:nth-of-type(2) {
  left: 2px;
  -webkit-animation: animation-1 0.6s infinite;
  animation: animation-1 0.6s infinite;
}

.emotion-1 div:nth-of-type(3) {
  left: 8px;
  -webkit-animation: animation-1 0.6s infinite;
  animation: animation-1 0.6s infinite;
}

.emotion-1 div:nth-of-type(4) {
  left: 14px;
  -webkit-animation: animation-2 0.6s infinite;
  animation: animation-2 0.6s infinite;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
